Cython/Compiler/ExprNodes.py

```diff
@@ -6764,18 +6810,22 @@ def attribute_is_likely_method(attr):
             keyword_variable = f", {self.kwdict.result()}"
         else:
             keyword_variable = ""
-        code.putln("%s = %s(%s, __pyx_callargs+1-%s, %d+%s%s);" % (
+        code.putln("%s = %s(%s, __pyx_callargs+1-%s, (%d+%s)%s %s);" % (
             self.result(),
             function_caller,
-            function,
+            code.get_py_string_const(self.function.attribute) if self.avoid_attribute_lookup else function,
```

Comment:
I'd make sure `function` is always assigned above and avoid the conditional usage here.